packages/core-mobile/app/store/balance/utils.ts

```diff
@@ -7,13 +7,20 @@ import { isTokenMalicious } from 'utils/isTokenMalicious'
 import { ChainId, Network } from '@avalabs/core-chains-sdk'
 import { getLastTransactedNetworks } from 'new/common/utils/getLastTransactedNetworks'
 import { uniqBy } from 'lodash'
+import { TokenType } from '@avalabs/vm-module-types'
 import { LocalTokenWithBalance } from './types'
 
 const UPDATE_PERIOD = 15
 
 export function getLocalTokenId(
   token: TokenWithBalance | NetworkContractToken
 ): string {
+  if (token.type === TokenType.SPL) {
+    return `spl-${token.address}`
+  }
+  if (token.type === TokenType.NATIVE && token.symbol === 'SOL') {
+    return 'native-sol'
+  }
```

Comment:
should we combine this logic with this one? also, let's use `_` instead of `-`. I think that is a bit safer. 
```   
    localId:  token.type === TokenType.SPL
                ? token.address
                : `SOL_${token.symbol}`,
```